File: Core/GameModels/GameState.cs
namespace Core.GameModels
{
    public enum GameStage
    {
        PreFlop,
        Flop,
        Turn,
        River,
        Showdown
    }

    public class GameState
    {
        public List<Player> Players { get; set; } = [];
        public int CurrentPlayerIndex { get; set; }
        public List<PlayerCard> CommunityCards { get; set; } = [];
        public List<PlayerCard> Deck { get; set; } = [];
        public int Pot { get; set; }
        public GameStage Stage { get; set; } = GameStage.PreFlop;
        public List<string> AvailableActions { get; set; } = [];
        public int BigBlind { get; set; }
        public int SmallBlind { get; set; }
        public int DealerPosition { get; set; }
        public int SmallBlindPosition { get; set; }
        public int BigBlindPosition { get; set; }
        public int HighestBet { get; set; } = 0;
        public bool IsGameOver { get; set; }
        public bool IsFirstRound { get; set; } = true;
        public List<int> WinnersPositions { get; set; } = [];
        public int TableId { get; set; }
        public DateTimeOffset StartedAt { get; set; }
        public int EarlyLeavePayout { get; set; }
    }
}

File: frontend/src/types/GameState.ts
export default interface Card {
    suit: string;
    rank: string;
    isHidden: boolean;
}

export interface Player { 
    name: string;
    chips: number;
    currentBet: number;
    hand: Card[];
    isFolded: boolean;
    isPlayer: boolean;
    isActive: boolean;
    isDealer: boolean;
    lastAction?: string;
    lastActionAmount?: number;
    userId?: string | null;
    profileImageUrl?: string | null;
}

export interface GameState {
    players: Player[];
    communityCards: Card[];
    deck: Card[];
    pot: number;
    stage: string;
    availableActions: string[];
    bigBlind: number;
    smallBlind: number;
    dealerPosition: number;
    smallBlindPosition: number;
    bigBlindPosition: number;
    currentPlayerIndex: number;
    highestBet: number;
    isGameOver: boolean;
    winnersPositions: number[];
    penaltyAmount: number;
    earlyLeavePayout: number;
    currentViewerUserId?: string | null;
}

export interface GameSessionSummary {
  duration: string;
  totalSeconds: number;
  startingChips: number;
  finalChips: number;
  profit: number;
  roundsPlayed: number;
  balanceReturned: number;
  penaltyAmount: number;
  wasEarlyLeave: boolean;
}

export type GameActionPayload = 
    | { amount: number }  
    | undefined               